teamA/lib/Views/login_page.dart

```diff
@@ -0,0 +1,189 @@
+            .primaryContainer,
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: [
+            // Title Section
+            Text(
+              'Welcome to Learning Lens Application!',
+              style: TextStyle(
+                fontSize: 24,
+                fontWeight: FontWeight.bold,
+                color: colorScheme
+
+                    // Text color based on the primaryContainer contrast color
+                    .onPrimaryContainer,
+              ),
+              textAlign: TextAlign.center,
+            ),
+            const SizedBox(height: 16),
+
+            // Instruction Section
+            Text(
+              'Please, enter your username and password below and click Login, to access the Dashboard.',
+              style: TextStyle(
+                fontSize: 16,
+                color: colorScheme
+
+                    // Text color for instructions
+                    .onPrimaryContainer,
+              ),
+              textAlign: TextAlign.center,
+            ),
+            const SizedBox(height: 32),
+
+            // Username Input
+            TextField(
+              controller: _usernameController,
+              decoration: InputDecoration(
+                labelText: 'Username',
+                hintText: 'Enter your username',
+                border: const OutlineInputBorder(),
+
+                // Input field background
+                fillColor: colorScheme.surface,
+                filled: true,
+              ),
+            ),
+            const SizedBox(height: 16),
+
+            // Password Input
+            TextField(
+              controller: _passwordController,
+              decoration: InputDecoration(
+                labelText: 'Password',
+                hintText: 'Enter your password',
+                border: const OutlineInputBorder(),
+
+                // Input field background
+                fillColor: colorScheme.surface,
+                filled: true,
+              ),
+              obscureText: true,
+            ),
+            const SizedBox(height: 16),
+
+            // Moodle URL (optional if needed)
+            TextField(
+              decoration: InputDecoration(
+                labelText: 'Moodle URL',
+                hintText: 'https://moodle.example.com',
+                border: const OutlineInputBorder(),
+
+                // Input field background
+                fillColor: colorScheme.surface,
+                filled: true,
+              ),
+            ),
+            const SizedBox(height: 32),
+
+            // Login Button
+            ElevatedButton(
+              onPressed: () async {
+                var wasSuccessful = await LoginScreen.controller.loginToMoodle(_usernameController.text, _passwordController.text);
+                if (wasSuccessful) {
+                Navigator.push(
```

Comment:
Might want to also pull the course list here like TestWizard did. I believe we talked about that during the meeting on Wednesday.

Reply:
![image](https://github.com/user-attachments/assets/12c97eb3-ea8d-4fe5-a143-a470335c53ba)
Will this suffice for now? Stored as a local variable but the controller can be called to retrieve the courses at any time after login. The usertoken is stored in the singleton

Reply:
![image](https://github.com/user-attachments/assets/9c0c49a5-fcbb-496c-a680-9fdeccd41add)